src/ophyd_async/core/_mock_signal_backend.py

```diff
@@ -13,7 +13,11 @@
 class MockSignalBackend(SignalBackend[SignalDatatypeT]):
     """Signal backend for testing, created by ``Device.connect(mock=True)``."""
 
-    def __init__(self, initial_backend: SignalBackend[SignalDatatypeT]) -> None:
+    def __init__(
+        self,
+        initial_backend: SignalBackend[SignalDatatypeT],
+        mock: bool | Mock = True,
```

Comment:
I think we should create the Mock before it gets passed in:
```suggestion
        mock: Mock,
```